modes/parkour/moderation.lua

```diff
@@ -0,0 +1,447 @@
+		end
+
+		showChatCommands(player, cmd, args)
+
+			if not tonumber(args[1]) then
+				requestplayer = capitalize(args[1])
+				if not string.find(requestplayer, "#", 1, true) then
+					requestplayer = requestplayer .. "#0000"
+				end
+			end
+
+			if quantity < 2 then 
+				if players_file[requestplayer] then 
+					local file = players_file[args[1]]
+					local getPlayerBanCount = file.bancount
+					tfm.exec.chatMessage("<v>[#] <r>"..args[1].. " have " ..getPlayerBanCount.. " bans in record.", player) 
+				else
+					ban_request[requestplayer] = {player, os.time() + 1000, "bancount"}
+					system.loadPlayerData(requestplayer)
+				end
+			return
+			end
+
+			if args[2] == "reset" then
+				if players_file[requestplayer] then 
+					local file = players_file[args[1]]
+					file.bancount = 0
+					savePlayerData(requestplayer)
+					tfm.exec.chatMessage("<v>[#] <r>"..args[1].. "'s bans has been reset", player) 
+				else
+					ban_request[requestplayer] = {player, os.time() + 1000, "resetbancount"}
+					system.loadPlayerData(requestplayer)
+				end
+			return
+		end 
+
+	elseif cmd == "setrank" then -- !setrank username ranks
+		if not ranks.admin[player] and not ranks.bot[player] then return end
+		if args[1] == player then return tfm.exec.chatMessage("<v>[#] <r>You can't change your rank.", player)  return end
+		local p = args[1]
+		local newranks = {}
+		local id = 0
+
+		if quantity < 2 then 
+			return tfm.exec.chatMessage("<v>[#] <r>!setrank username ranks", player) 
+		end
+
+		if args[2] ~= "none" then
+			for i=2, #args do
+				if ranks_id[args[i]] then
+					id = id + ranks_id[args[i]]
+					newranks[args[i]] = true
+				else
+					return tfm.exec.chatMessage("<v>[#] <r>Invalid rank: "..args[i] , player)
+				end
+			end
+		end
+
+		if id == 0 then
+			player_ranks[p] = nil
+			id = nil
+			tfm.exec.chatMessage("<v>[#] <r>All ranks removed from "..p , player)
+		else
+			player_ranks[p] = newranks
+			tfm.exec.chatMessage("<v>[#] <r>Ranks has been set." , player)
+		end
+
+		schedule("modify_rank", p, id)
+end end)
+
+onEvent("PlayerDataParsed", function(player, data)
+	checkBanRequest(player, data)
+end)
+
+onEvent("OutPlayerDataParsed", checkBanRequest)
+
+
+onEvent("Loop", function(elapsed)
+	local now = os.time()
+
+	local to_remove, count = {}, 0
+	for player, data in next, ban_request do
+		if now >= data[2] then
+			count = count + 1
+			to_remove[count] = player
+			translatedChatMessage("cant_load_profile", data[1], player)
+		end
+	end
+
+	for idx = 1, count do
+		ban_request[to_remove[idx]] = nil
+	end
+end)
+
+onEvent("GameStart", function()
+	system.disableChatCommandDisplay("ban")
+	system.disableChatCommandDisplay("unban")
+	system.disableChatCommandDisplay("pban")
+	system.disableChatCommandDisplay("punban")
+	system.disableChatCommandDisplay("addmap")
+	system.disableChatCommandDisplay("removemap")
+	system.disableChatCommandDisplay("bancount")
+	system.disableChatCommandDisplay("setrank")
```

Comment:
Maybe we could simply use `system.disableChatCommandDisplay(nil)` and remove all those nasty function calls?